src/sage/combinat/permutation.py

```diff
@@ -5270,49 +5270,52 @@ def shifted_shuffle(self, other):
         right_permutohedron_interval(self.shifted_concatenation(other, "left"))
 
 
-    def kth_roots(self, k=2):
+    def nth_roots(self, n):
         r"""
-        Return all k-th roots of self (as a generator).
+        Return all n-th roots of ``self`` (as a generator).
         
-        A k-th root of the permutation self is a permutation Gamma such that Gamma^k == self.
+        A n-th root of the permutation ``self`` is a permutation `\gamma` such that `\gamma^n == self`.
 
-        Note that the number of k-th roots only depend on the cyclic type of `self`.
-        
-        INPUT:
-        
-        - k -- optional integer (default 2), at least 1
+        Note that the number of n-th roots only depend on the cyclic type of ``self``.
         
         EXAMPLES::
         
             sage: Sigma = Permutations(5).identity()
-            sage: list(Sigma.kth_roots(3))
+            sage: list(Sigma.nth_roots(3))
             [[1, 4, 3, 5, 2], [1, 5, 3, 2, 4], [1, 2, 4, 5, 3], [1, 2, 5, 3, 4], [4, 2, 3, 5, 1], [5, 2, 3, 1, 4], [3, 2, 5, 4, 1], [5, 2, 1, 4, 3], [2, 5, 3, 4, 1], [5, 1, 3, 4, 2], [2, 3, 1, 4, 5], [3, 1, 2, 4, 5], [2, 4, 3, 1, 5], [4, 1, 3, 2, 5], [3, 2, 4, 1, 5], [4, 2, 1, 3, 5], [1, 3, 4, 2, 5], [1, 4, 2, 3, 5], [1, 3, 5, 4, 2], [1, 5, 2, 4, 3], [1, 2, 3, 4, 5]]
             
             sage: Sigma = Permutation('(1, 3)')
-            sage: list(Sigma.kth_roots())
+            sage: list(Sigma.nth_roots(2))
             []
         
-        For n >= 6, this algorithm begins to be more efficient than naive search (look at all permutations and test its k-th power).
+        For n >= 6, this algorithm begins to be more efficient than naive search (look at all permutations and test their n-th power).
         
         .. SEEALSO::
         
-                * :meth:`has_kth_root`
-                * :meth:`number_of_kth_roots`
+                * :meth:`has_nth_root`
+                * :meth:`number_of_nth_roots`
```

Comment:
You also may want to compare with the functionality provided by `GroupLibGap`, also speed-wise.  For example:

```
sage: G = GroupLibGAP(SymmetricGroup(5)._libgap_())
sage: G.one().nth_roots(3)
[1,
 (1,2,3),
 (2,3,5),
 (1,5,3),
 (1,2,5),
 (2,5,3),
 (2,3,4),
 (1,3,5),
 (1,4,3),
 (1,5,2),
 (1,2,4),
 (1,3,2),
 (2,4,3),
 (3,4,5),
 (2,5,4),
 (1,3,4),
 (3,5,4),
 (1,4,5),
 (1,4,2),
 (2,4,5),
 (1,5,4)]
```

Reply:
For identity permutation of size 8, computing cubic roots (i.e. `Permutations(8).identity().nth_roots(3)`) takes 4.56 micro on my PC, versus 5.73 ms with GAP. We gain a factor 1000.
How do I indicate this in the docstring? (More generaly, how and when do we put speed test in doc?)

Reply:
There is no formalized way to do it.  Personally, I like to write a test in the TESTS section which is just fast enough not to be annoying when running the tests (about half a second or so, and so that using the old method (if there is any) one would notice that the test takes too long.